### PR TITLE
fix: getting rid of unnecessary guid-typescript dependency

### PR DESCRIPTION
**Pull request by markovav-official**
Fixing PR [microsoft/kiota-typescript#1154](https://github.com/microsoft/kiota-typescript/pull/1154)

Fixes [microsoft/kiota#4508](https://github.com/microsoft/kiota/issues/4508)
Using a custom Guid type to get rid of unnecessary package (and partly broken) guid-typescript.

### CODE REVIEW

**Comment by markovav-official** [comment]
@microsoft-github-policy-service agree

**Comment by andrueastman** [comment]
Thanks for creating this @markovav-official 

Just to confim, as there are still references to the package in the generation side, will you still be authoring another PR to clean that up as well? 
See example reference at https://github.com/microsoft/kiota/blob/04815fe41df8457b7ac4a957f1eece3ab7c9884f/src/Kiota.Builder/Refiners/TypeScriptRefiner.cs#L695

**Comment by markovav-official** [comment]
Hello! Thank you for your reply and mention on the connected issue.
I made an additional PR: [microsoft/kiota#5734](https://github.com/microsoft/kiota/pull/5734)

**Review by andrueastman — APPROVED** [review]
Thanks for the contribution @markovav-official

File: packages/abstractions/src/utils/guidUtils.ts [hunk 1]
```diff
@@ -4,17 +4,38 @@
...
+ * Generates a part of a GUID.
+ * @param count the number of 2 byte chunks to generate.
+ * @returns a part of a GUID.
  */
-export function parseGuidString(source?: string): Guid | undefined {
-	if (source && Guid.isGuid(source)) {
-		return Guid.parse(source);
-	} else {
-		return undefined;
+export const gen = (count: number) => {
```
Comment (baywet) [inline packages/abstractions/src/utils/guidUtils.ts:34]: can you provide more context on why does this function need to be exported?

**Review by baywet — COMMENTED** [review]
Thanks for the contribution! Joining the party late since I was out of office.

File: packages/abstractions/src/utils/guidUtils.ts (hunk 1 above)
Comment (markovav-official) [reply packages/abstractions/src/utils/guidUtils.ts:34]: Hello! I agree, exporting here makes no sense, should I create a PR to delete it?

Comment (baywet) [reply packages/abstractions/src/utils/guidUtils.ts:34]: Please go ahead

Comment (baywet) [reply packages/abstractions/src/utils/guidUtils.ts:34]: Please go ahead

Comment (markovav-official) [reply packages/abstractions/src/utils/guidUtils.ts:34]: Done, https://github.com/microsoft/kiota-typescript/pull/1481

**Pull request merged**